Build.bee.cs

```diff
@@ -0,0 +1,377 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Bee.BuildTools;
+using Bee.Core;
+using Bee.Core.Stevedore;
+using Bee.NativeProgramSupport;
+using Bee.Toolchain.Linux;
+using Bee.Toolchain.LLVM;
+using Bee.Toolchain.MacOS;
+using Bee.Toolchain.VisualStudio;
+using Bee.Toolchain.Windows;
+using Bee.Tools;
+using NiceIO;
+
+class Build
+{
+    static void Main()
+    {
+        using var _ = new Bee.Core.BuildProgramContext();
+
+        Backend.Current.StevedoreSettings = new StevedoreSettings
+        {
+            Manifest = { "manifest.stevedore" },
+            EnforceManifest = true,
+        };
+
+        foreach (var sdk in Sdks.ForCurrentPlatformEditorSdks())
+        {
+            foreach (CodeGen codeGen in Enum.GetValues(typeof(CodeGen)))
+            {
+                var programConfiguration = new ProgramConfiguration(sdk, codeGen);
+                SetupLib(programConfiguration);
+                SetupStevedoreArtifact(programConfiguration);
+            }
+        }
+    }
+
+    enum CodeGen
+    {
+        Debug,
+        Release
+    }
+
+    class ProgramConfiguration
+    {
+        public ProgramConfiguration(Sdk sdk, CodeGen codeGen)
+        {
+            Sdk = sdk;
+            CodeGen = codeGen;
+        }
+
+        public string GetTargetName(string separator = "_")
+        {
+            var targetNameParts = new List<string> {
+                Sdk.Platform.DisplayName.ToLower(),
+                Sdk.Architecture.DisplayName
+            };
+
+            if (CodeGen == CodeGen.Debug)
+            {
+                targetNameParts.Add("dbg");
+            }
+
+            return string.Join(separator, targetNameParts);
+        }
+
+        public string GetArtifactName(string separator = "_")
+        {
+            var artifactNameParts = new List<string> {
+                "tbb",
+                GetTargetName(separator: separator)
+            };
+
+            return string.Join(separator, artifactNameParts);
+        }
+
+        public Sdk Sdk { get; private set; }
+        public CodeGen CodeGen { get; private set; }
+    }
+
+    static void SetupLib(ProgramConfiguration programConfiguration)
+    {
+        var cfg = CfgFor(programConfiguration.CodeGen);
+        var targetId = programConfiguration.GetTargetName();
+
+        NPath[] inputs;
+        var commandLineArguments = new List<string>
+        {
+            $"arch={ArchFor(programConfiguration.Sdk.Architecture)}",
+            $"compiler={CompilerFor(programConfiguration.Sdk)}",
+            $"cfg={cfg}",
+            $"tbb_build_prefix={targetId}",
+            $"-j {Environment.ProcessorCount}",
```

Comment:
I think this deserves a comment, because it's usually a bad idea to run things like `make -j$(nproc)` from the build graph where actions are already being run in parallel. I assume that you understand the issue and this outer wrapper "bee" build is small enough for this not to be a problem, but it would be good to leave a warning note here for future readers.